### PR TITLE
Yarn scripts

### PR DESCRIPTION
**Pull request by Smokyish**
- Updates README with the new script info
- Adds yarn scripts to setup, update and deploy changes from external repos
- Updates sidebar to use new content pulled from external repos

There are some issues with this:
- Some content in repos is using `.rst`, they need to be changed to MarkDown in order to work
- Contents in repos need to be changed to remove duplicate titles etc.

Please clone and run locally to see the changes and make changes in the repos before merging this.

### CODE REVIEW

**Comment by cducrest** [comment]
I don't get it, are we supposed not to have duplicates titles of files for documentation across repos? Can this not be avoided by placing what is pulled in a folder with the name of the repo / directory from where it is pulled?

**Comment by Smokyish** [comment]
@cducrest You have the option to either make the title change in the repo MarkDown file, or in the script that injects the `title` into the doc here in the dev-docs. The layout injection is required for things to work in dev-docs though.

File: docs/blockchain/repo/tlbc.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
+---
```
Comment (berndbohmeier) [inline docs/blockchain/repo/tlbc.md:1]: These files are generated right. This should be written somewhere, best in the folder or file

File: package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
...ustlines Smart Contract Platform''\n''---''\n' | cat - docs/contracts/repo/contracts_README.md > temp && mv temp docs/contracts/repo/contracts_README.md && cp submodules/relay/README.rst docs/relay/repo/relay_server.md && echo '---''\n''layout: relay''\n''title: Trustlines relay server''\n''---''\n' | cat - docs/relay/repo/relay_server.md > temp && mv temp docs/relay/repo/relay_server.md && cp submodules/relay/docs/RelayAPI.md docs/relay/repo/relay-api.md && echo '---''\n''layout: relay''\n''title: Trustlines Relay API''\n''---''\n' | cat - docs/relay/repo/relay-api.md > temp && mv temp docs/relay/repo/relay-api.md && cp submodules/relay/docs/RelayServer.md docs/relay/repo/running_trustlines_system.md && echo '---''\n''layout: relay''\n''title: Running the Trustlines system''\n''---''\n' | cat - docs/relay/repo/running_trustlines_system.md > temp && mv temp docs/relay/repo/running_trustlines_system.md && cp submodules/relay/docker/trustlines/README.md docs/relay/repo/trustlines_docker.md && echo '---''\n''layout: relay''\n''title: Running Trustlines via docker-compose''\n''---''\n' | cat - docs/relay/repo/trustlines_docker.md > temp && mv temp docs/relay/repo/trustlines_docker.md",
```
Comment (berndbohmeier) [inline package.json:10]: ehm, this is unreadable. 
Is this really the only option?
Lets put it at least in an own file with proper formatting.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "update": "git pull --recurse-submodules && yarn build",
+    "clean": "rm docs/blockchain/repo/*.md && rm docs/clientlib/repo/*.md && rm docs/contracts/repo/*.md && rm docs/relay/repo/*.md",
+    "deploy": "git checkout gh-pages && git merge master && git push && git checkout master"
```
Comment (berndbohmeier) [inline package.json:13]: There is a tool for this. https://www.npmjs.com/package/gh-pages

**Comment by Smokyish** [comment]
@berndbohmeier 

> Why do we copy all the files, I thought the goal of the submodules was so that we do not have to do that?

The expected result was that Jekyll would generate the content from the files in the submodules folders, but it turned out that that was not the case. So options were to either copy the files into the docs folders or write a new generator for that, which might not had been as quick and easy, so this was the solution done to fix that and get the content generated and visible.

**Pull request merged**